src/hipercow/resources.py

```diff
@@ -0,0 +1,158 @@
+"""Specify and interact with resources."""
+
+from dataclasses import dataclass
+
+
+@dataclass(kw_only=True)
+class TaskResources:
+    """Resources required for a task.
+
+    We don't support everything that the R version does yet; in
+    particular we've not set up `hold_until`, `priority` or
+    `requested_nodes`, as these are not widely used.
+
+    Attributes:
+        queue: The queue to run on.  If not given (or `None`), we use
+            the default queue for your cluster.  Alternatively, you
+            can provide `.default` for the default queue or `.test`
+            for the test queue.
+
+        cores: The number of cores to request.
```

Comment:
Tempting to add: 

"Note that increasing the number of cores only gives you a performance improvement if your code is able to use those cores."

Reply:
good idea, I've added something along these lines